gems/aws-sdk-core/lib/aws-sdk-core/plugins/regional_endpoint.rb

```diff
@@ -150,7 +150,18 @@ def resolve_endpoint(cfg)
           # that a custom endpoint has NOT been configured by the user
           cfg.override_config(:regional_endpoint, true)
 
-          resolve_legacy_endpoint(cfg)
+          # preserve legacy (pre EP2) client.config.endpoint still resolves a value
+          # when needed.
+          struct = cfg.instance_variable_get(:@struct)
+          if struct
+            # need a separate proc to get namespace/private access to resolve_legacy_endpoint
```

Comment:
Should the method just be public to the plugin then?

Reply:
what do you mean by public to the plugin?  I can add a public accessor for struct to ConfigResolver

Reply:
Check the most recent commit - I've added an accessor for struct and cleaned this code up.  Let me know if thats what you were thinking